x-pack/plugins/cases/server/client/metrics/actions/aggregations/isolate_host.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IsolateHostActionType } from '../../../../../common/api';
+import { CASE_COMMENT_SAVED_OBJECT } from '../../../../../common/constants';
+import { AggregationBuilder, AggregationResponse } from '../../types';
+
+interface ActionsAggregation {
+  actions?: {
+    buckets: Array<{ key: string; doc_count: number }>;
+  };
+}
+
+export class IsolateHostActions implements AggregationBuilder {
+  constructor(private readonly uniqueValuesLimit: number = 10) {}
+
+  build() {
+    return {
+      actions: {
+        terms: {
+          field: `${CASE_COMMENT_SAVED_OBJECT}.attributes.actions.type`,
```

Comment:
Couple thoughts about the terms aggregation:

We could instead explicitly create a cardinality aggregation for both type `isolated` and `unisolated` (in a single query). I believe terms aggregations are great when we don't know the values of the things we're aggregating over (like ids etc).

The advantage of doing it with a terms aggregation though is if we add a new type like "almost isolated" haha we won't have to update the query where we would if we used a cardinality aggregation. So either way it's up to you haha.

Reply:
I think the cardinality agg is not what we need here, it returns the count of different values a field has:
[docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-cardinality-aggregation.html):
> A single-value metrics aggregation that calculates an approximate count of distinct values.

I can imagine that a cardinality would make sense if we wanted to ignore the actions that are performed to the same host more than once, so we could count the distinct hosts for each `action.type`. But I think the `total` we need here are just value occurrences for each `action.type`, which is not the same as the `total` for the hosts and users aggregations, that are a count of distinct values.

Reply:
Yeah you're right, a cardinality isn't quite right. I suppose we'd have to do two separate filter aggregations for `action.type: isolated` and `action.type: unisolated` and then do a count of the resulting documents. What you have is fine 👍